modules/Quiz/resources/IVQModule.js

```diff
@@ -360,22 +362,24 @@
              * @param {*} openQuestionText 
              */
             submitAnswer:function(questionNr,selectedAnswer,openQuestionText){
-                var _this = this,isAnswered;
+                var _this = this;
+                var defer = $.Deferred();
                 $.cpObject.cpArray[questionNr].selectedAnswer = selectedAnswer;
-                if ($.cpObject.cpArray[questionNr].isAnswerd) {
-                    isAnswered = true;
-                }
-                else{
-                    isAnswered = false;
-                    $.cpObject.cpArray[questionNr].isAnswerd = true;
-                }
-                _this.KIVQApi.addAnswer(isAnswered,_this.i2q(selectedAnswer),_this.kQuizUserEntryId,questionNr,function(data){
+                $.cpObject.cpArray[questionNr].submitRequested = true;
+                var submitCallTimeout = setTimeout(function(){
+                    defer.reject();
+                }, _this.quizPlugin.getConfig("apiRequestTimeout") || 5000)
+                _this.KIVQApi.addAnswer($.cpObject.cpArray[questionNr].isAnswerd,_this.i2q(selectedAnswer),_this.kQuizUserEntryId,questionNr,function(data){
                     if (!_this.checkApiResponse('Add question err -->',data)){
-                        return false;
-                    }else {
+                        defer.reject();
+                    } else {
+                        clearTimeout(submitCallTimeout);
+                        $.cpObject.cpArray[questionNr].isAnswerd = true;
                         $.cpObject.cpArray[questionNr].answerCpId = data.id;
+                        defer.resolve();
```

Comment:
@semarche-kaltura  we still need to think if we can handle success callback after timeout. very edge case where the BE responses really slow but still possible. 
I will try to think with Eran on this.

Reply:
We have a BE request to return an BE error when we try to submit the same question twice. Currently it is allowed 
https://kaltura.atlassian.net/browse/PLAT-10458